### ISSUE TITLE
allow the user to associate personal help text with a window or control

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#13977]
### Describe the solution you'd like
my suggestion is that nvda allows the user to type some text and associate this text with an open control or window.
For example, let's imagine that the user has difficulty remembering that when he arrives at the work area he must scroll with the arrows up or down, as well as right and left arrows.
so the user could tell nvda every time it arrives on the desktop to say a message that the user himself has typed, for example, on the desktop also navigate with right and left arrows.
Another example, the user could tell nvda that every time word is opened the reader tells him to type ctrl + n for bold, ctrl + i for italics.
example portuguese brazil.
To reinforce, the help text is the user who is going to type it, a personal reminder to him to do something in a specific window or control.
nvda will not provide this suggestion nor a help text, although I would like this in other previously opened problems.
I can also use as a justification for this feature that a text written by the user makes it easier for him to remember doing some activity.
I thought of this resource when faced daily in my work of teaching users with learning difficulties or very old people how to use the computer.
I think of a simple screen for this feature:
a radio button where the user chooses whether help text will be spoken when opening the currently focused window or the currently focused control.
and a text field where the user can type their personal message for the occasion.
To make the implementation simple at first, no allowing messages for types of controls depending on each program, nor allowing messages for different documents, although this can be explored in the future.
### Describe alternatives you've considered
I accept different window layouts than the one I sited, or even more restrictions to further simplify the implementation, as long as the main idea in this feature could be done.
### Additional context
nvda is an important altonomy tool for people with difficulties, but the reader can make this altonomy more comfortable to the user by providing tools to help him work in the computer world.
Users often do not complete an activity because of forgetting as simple as not pressing an arrow in a specific location.
Older users have difficulty understanding these activities and are able to memorize better when listening to their own explanation of how to interact with a specific window or control.

**Comment by fernando-jose-silva** [comment]
Sorry, I remembered another reason.
Teachers could also customize copies of students' nvda with text messages to help them develop certain activities, myself as a teacher would use this.
The teacher is close to his students, and knows where their difficulties are greatest.
The user could also easily change these messages or deactivate them if they no longer need them.

**Comment by josephsl** [comment]
CC @NVDAEs, is this something we can try with Control Usage Assistant first? This is an advanced form of #2699. Thanks.

**Comment by nvdaes** [comment]
Yes. If you want, create a PR in nvdaes/controlUsageAssistant repo or I"ll try to do it.

Enviado desde mi iPhone

**Comment by fernando-jose-silva** [comment]
@josephsl 
thank you very much for your always attention.

@nvdaes 
Thank you very much for your work, I would like to let you know that the automatic helps you have implemented in the controlUsageAssistant-
it's used by me in classes, and it really helps students with difficulties, and I'm sure that if I manage to implement personalized user help as well, this will leverage the students' learning a lot.

**Comment by fernando-jose-silva** [comment]
if this implementation works, it would be great to have it natively in nvda

**Comment by nvdaes** [comment]
For implementation in controlUsageAssistant, I've opened this [discussion](https://github.com/nvdaes/controlUsageAssistant/discussions/10), so we can use this issue for NVDA.

**Comment by seanbudd** [comment]
Is this a duplicate of #2111 or #12488?

**Comment by Adriani90** [comment]
@seanbudd this is rather another design proposal related to #2699. But as far as I understand this design proposal of automatic saying the help messages when focusing the coresponding area has many advantages compared to pressing f1 or any other command to hear the context sensitive help message.
Also this design request proposes that the user should be able to customize the help messages, although I am rather in favor of taking the dictionary of help messages already defined in the control ussage assistant addon and make NVDA report the coresponding message automatically when focusing the certain area or element. This gives much smoother user experience and eliminates the risk that a lot of non sense help messages are created.

The issues you referenced are about changing the label of an element to be different than what is displayed visually on the screen, so no duplicate here.

**Comment by fernando-jose-silva** [comment]
When I made this request, and emphasized that the user could type a personalized text, I did this thinking from the point of view of a teacher, this is my daily life, I am a computer teacher, and a cell phone teacher, for other people who have disabilities visual, in an institution, or giving private classes.
I thought, for example, that I could give a very brief explanation of how, for example, some basic shortcuts for Word, such as bold, italics, among others, as soon as the user enters Word, as well as in other programs, such as PowerPoint, such as o press five to open slideshow.
I understand that these are unwanted announcements for the vast majority of users, therefore, I would place these teacher announcements for the computers where I teach, or for some users who wish, but they would have the autonomy to turn off these settings as soon as they are ready. more capable.
remembering, my students are people who have difficulty dealing with computers, very elderly people, up to 95 years old, children who are just beginning to use this type of tool, people with other intellectual impairments.
but thinking about the possibility of this being a facilitator for major problems, we could think about leaving these settings in a difficult to access place, where only people who really know the screen reader can access
Don't demand too much, for example I'm a teacher, but I'm not a programmer, if this requires me to change the screen reader code, I won't be able to do it.
